tests/torch/pruning/test_model_pruning_analysis.py

```diff
@@ -170,23 +170,45 @@ def __init__(self, model: Union[Type[torch.nn.Module], Callable[[], torch.nn.Mod
                                                    8: PruningAnalysisDecision(
                                                           False, [PruningAnalysisReason.LAST_CONV])},
                                   prune_params=(True, False)),
-    GroupPruningModulesTestStruct(model=TestModelResidualConnection,
-                                  non_pruned_module_nodes=['TestModelResidualConnection/NNCFConv2d[conv4]/conv2d_0',
-                                                      'TestModelResidualConnection/NNCFConv2d[conv5]/conv2d_0'],
+    GroupPruningModulesTestStruct(model=partial(TestModelResidualConnection, last_layer_accept_pruning=False),
+                                  non_pruned_module_nodes=['TestModelResidualConnection/NNCFLinear[linear]/linear_0',
+                                                           'TestModelResidualConnection/NNCFConv2d[conv4]/conv2d_0',
+                                                           'TestModelResidualConnection/NNCFConv2d[conv5]/conv2d_0'],
                                   pruned_groups=[['TestModelResidualConnection/NNCFConv2d[conv1]/conv2d_0',
                                                   'TestModelResidualConnection/NNCFConv2d[conv2]/conv2d_0',
                                                   'TestModelResidualConnection/NNCFConv2d[conv3]/conv2d_0']],
                                   pruned_groups_by_node_id=[[1, 2, 4]],
                                   can_prune_after_analysis={0: True, 1: True, 2: True, 3: True, 4: True,
-                                                            5: True, 6: True, 7: False, 8: False, 9: False,
-                                                            10: False, 11: False, 12: False},
+                                                            5: True, 6: True, 7: True, 8: True, 9: True,
+                                                            10: True, 11: True, 12: True},
```

Comment:
Now the test cases are vulnerable against a bug in the logic when the code would mistakenly report every node is reported as "prunable". Make sure that there are enough test cases where some of the nodes are actually not prunable as the reference results.

Reply:
Agee, now count of prunable layers is increased, so I've added test with mobilenet_v3 with mean instead of avgpool layer. Here model analysis should indicate all nodes is unprunalbe. New test is covering thins